### PR TITLE
build!: use rollup

### PR DESCRIPTION
**Pull request by jinlee93**
### Summary:
- uses rollup to build EDS and its stylesheet
- revives Huth's rollup pr #1431 
- TODO:
  - [x] Resolve circular dependencies: #1557
image
  - [x] Resolve unresolved dependencies: #1558 
image
  - [x] ~~CSS post processing~~ remix / downstream apps will process
  - [x] Provide CJS as well #1572
  - [x] Clean up unused tokens #1581 

### Test Plan:

- [ ] Wrote [automated tests](https://czi.atlassian.net/wiki/x/Hbl1H)
- [ ] CI tests / new tests are not applicable
- [ ] Manually tested my changes, but I want to keep the details secret
- [x] Manually tested my changes, and here are the details:
  - Created alpha publish and testing on edu-stack: https://github.com/chanzuckerberg/edu-stack/pull/68
- [x] edu stack tests pass with alpha publish
  - No visual regressions [running chromatic locally](https://www.chromatic.com/build?appId=64384fda22cb8f0811c02bb3&number=4)

### CODE REVIEW

File: src/index.ts [hunk 1]
```diff
@@ -1,3 +1,5 @@
+import './tokens-dist/css/variables.css';
```
Comment (jinlee93) [inline src/index.ts:1]: this brings in the tokens that are css custom properties into the built stylesheet

File: tsconfig.json [hunk 2]
```diff
@@ -4,17 +4,17 @@
...
     "lib": ["es5", "es6", "dom"],
     "moduleResolution": "node",
     "noUnusedLocals": true,
-    "outDir": "lib",
     "resolveJsonModule": true,
     "skipLibCheck": true,
     "strict": true,
     "suppressImplicitAnyIndexErrors": true,
-    "target": "es6"
+    "target": "es2018"
```
Comment (ahuth) [inline tsconfig.json:18]: All our supported browsers from https://go.czi.team/edu-browser-support support es2018, so this is 👍

File: .github/workflows/test.yml [hunk 3]
```diff
@@ -41,6 +41,14 @@ jobs:
       - name: Build Dist 🔧
         run: yarn run build
 
+      - name: Check for "tslib" in build # tslib is peer dep of rollup ts plugin, but we're not including tslib due to lack of need
```
[the remark this answers is not in the document]
Comment (jinlee93) [reply .github/workflows/test.yml:44]: That sounds great, replaced the comment

**Review by ahuth — APPROVED** [review]
LGTM 🤘

File: tsconfig.json (hunk 2 above)
Comment (booc0mtaco) [reply tsconfig.json:18]: @ahuth @jinlee93 I think that `suppressImplicitAnyIndexErrors` may have come back in a conflict resolution. We should be good in the repo. tho. Will create a task to remove this one line

**Pull request merged**